src/main/java/bio/terra/stairway/Stairway.java

```diff
@@ -897,11 +900,19 @@ void recoverReady()
       throws DatabaseOperationException, InterruptedException, StairwayExecutionException {
     List<String> readyFlightList = flightDao.getReadyFlights();
     for (String flightId : readyFlightList) {
-      if (workQueueEnabled) {
-        FlightContext flightContext = flightDao.makeFlightContextById(flightId);
-        queueFlight(flightContext);
-      } else {
-        resume(flightId);
+      try {
+        if (workQueueEnabled) {
+          FlightContext flightContext = flightDao.makeFlightContextById(flightId);
+          queueFlight(flightContext);
+        } else {
+          resume(flightId);
+        }
+      } catch (MakeFlightException | JsonConversionException e) {
+        // Catch only exceptions that are unlikely to be resolved with retries. Allow the rest of
+        // recovery to continue.
+        logger.error(String.format("Unable to recover flight id %s", flightId), e);
+        // Mark the flight as having fatally failed so it does not stay un-recovered forever.
+        flightDao.fatalFail(flightId, e);
```

Comment:
`fatalFail` method may also throw exceptions, and that would also cause Stairway fail to initialize. 
Do you think we should initialize the flight when that happens or just leave it in RUNNING state, and hope next time recover call can pick it up?

Reply:
Good point, added one last catch.
We already tried to initialize the flight, so we're pretty much out of options at this point and have to leave it in a running state. We can hope the next recover can pick it up, but we should really have alerting if this rare case starts to happen.